### ISSUE TITLE
optional value in bureaucracy struct_field is not used

### ISSUE THREAD

**Issue by jeremie-pierson** [cosmocode/dokuwiki-plugin-struct#283]
When using the struct_field field type in bureaucracy, field values are not used :
```
     <form>
     Action template templates:product product :
     struct_field "products.product" "=Some Value" @
     submit "Create new product"
     </form>
```

Here the value "Some Value" should be pre-filled in the form, but is not.

I think that it's caused by the `initialize` method overridden in `helper/field.php`. It calls the parent method, which in turn calls `init` and `standardArgs`. `standardArgs` calls `setVal()` if a value is defined. But the code in `helper_plugin_struct_field->setVal()` expects the `column` field to be set, and it is only set after the call to `parent::initialize()`.

Maybe the `init()` method should be overridden instead : it would allow the column to be set before `setVal` is called in `parent::initialize` :

``` diff
diff --git a/lib/plugins/struct/helper/field.php b/lib/plugins/struct/helper/field.php
index 961478f..5d3ac42 100644
--- a/lib/plugins/struct/helper/field.php
+++ b/lib/plugins/struct/helper/field.php
@@ -21,8 +21,8 @@ class helper_plugin_struct_field extends helper_plugin_bureaucracy_field {
      *
      * @param array $args
      */
-    public function initialize($args) {
-        parent::initialize($args);
+    protected function init(&$args) {
+        parent::init($args);
 
         // find the column
         try {
```

**Comment by iainhallam** [comment]
Would this also mean you could link to a page with such a form in and prefill the struct field via the link, as with other bureaucracy fields?

**Comment by annda** [comment]
This has been fixed in 89be912b2828f73aa728ab2d1347de9d26aada17

**Comment by jeremie-pierson** [comment]
Thanks! Will test when we upgrade our main internal Dokuwiki.
